### PR TITLE
Ink/spa segmented control

### PR DESCRIPTION
**Pull request by avh4**
SegmentedControl.V7 supports SPA links,  and also applies a couple UI improvements (new color for selected tab text, state change transition time, hover state)  and adds use of `aria-current`.

And updates the style guide example:
Screen Shot 2019-06-04 at 4 01 15 PM

> Hello there, wonderful author of widgets!
> This is `Nri.Ui.Friendly.AI.V1` speaking to you!
> Are you looking to get this awesome work reviewed as quickly as possible, so you can start putting it to use?
> Then feel free to ask Teambot to assign a reviewer to your PR
>
> https://nri-teambot.herokuapp.com
>
> It's best if we have separate pull requests for code changes and package version bumps. If you're just changing code, great! If you're just bumping the version, check out https://github.com/NoRedInk/noredink-ui#deploying.

### CODE REVIEW

**Comment by quelledanielle** [comment]
Please expose.

File: src/Nri/Ui/SegmentedControl/V7.elm [hunk 1]
```diff
@@ -64,15 +78,31 @@ tabList =
...
+                Nothing ->
+                    -- This is for a non-SPA view
+                    Html.button
+                        (Events.onClick (config.onClick option.value)
+                            :: attrs
+                        )
+                        children
+
+                Just toUrl ->
+                    -- This is a for a SPA view
```
Comment (tesk9) [inline src/Nri/Ui/SegmentedControl/V7.elm]: Love the comments, thanks!!!

File: src/Nri/Ui/SegmentedControl/V7.elm [hunk 2]
```diff
@@ -102,22 +102,25 @@ viewTab maybeToUrl config option =
...
-                unFocusedTabStyles
-        , css <|
-            case config.width of
+        (List.concat
+            [ [ Role.tab
+              , css sharedTabStyles
+              ]
+            , if option.value == config.selected then
+                [ css focusedTabStyles
+                , attribute "aria-current" "page"
```
Comment (tesk9) [inline src/Nri/Ui/SegmentedControl/V7.elm]: 🎨 We could use https://package.elm-lang.org/packages/tesk9/accessible-html-with-css/latest/Accessibility-Styled-Aria#currentPage here if we wanted

File: src/Nri/Ui/SegmentedControl/V7.elm [hunk 3]
```diff
@@ -68,8 +67,7 @@ tabList =
 viewTab : Config a msg -> Option a -> Html.Html msg
 viewTab config option =
     Html.div
-        [ Attr.id option.id
```
[the remark this answers is not in the document]
Comment (avh4) [reply src/Nri/Ui/SegmentedControl/V7.elm]: ah, okay -- I think if we add that in the future, we could automatically generate the ids based on the label -- which we do in some other modules already.  does that sound okay?

File: src/Nri/Ui/SegmentedControl/V7.elm [hunk 4]
```diff
@@ -53,7 +54,20 @@ type alias Icon =
...
+
+
+{-| Creates a segmented control that supports SPA navigation.
+You should always use this instead of `view` when building a SPA
+and the segmented control options correspond to routes in the SPA.
+
+The first parameter is a function that takes a `route` and returns the URL of that route.
+
+-}
+viewSpa : (route -> String) -> Config route msg -> Html msg
```
[the remark this answers is not in the document]
Comment (avh4) [reply src/Nri/Ui/SegmentedControl/V7.elm]: ... but it's Nri.Ui.... not NRI.UI.... ❓

File: src/Nri/Ui/SegmentedControl/V7.elm (hunk 3 above)
[the remark this answers is not in the document]
Comment (tesk9) [reply src/Nri/Ui/SegmentedControl/V7.elm]: Yeah, for sure! As long as there's some id-based association between the tab panel and the tab, I think we're set. :)

**Comment by avh4** [comment]
> by the segmented control uses the tabpanel role

@tesk9 should we have this view take `content : Html msg` as a parameter, so that we can always add the tabpanel div?

File: src/Nri/Ui/SegmentedControl/V7.elm (hunk 4 above)
[the remark this answers is not in the document]
Comment (tesk9) [reply src/Nri/Ui/SegmentedControl/V7.elm]: ...and Html and not HTML. 🤷‍♀ Spa is fine. I don't feel passionately about it.

[the remark this answers is not in the document]
Comment (avh4) [reply src/Nri/Ui/SegmentedControl/V7.elm]: (Also, other modules, like Button.V8 currently use "Spa", not "SPA")

File: elm.json [hunk 5]
```diff
@@ -50,6 +50,7 @@
         "Nri.Ui.PremiumCheckbox.V3",
         "Nri.Ui.PremiumCheckbox.V4",
         "Nri.Ui.SegmentedControl.V6",
+        "Nri.Ui.SegmentedControl.V7",
```
Comment (quelledanielle) [inline elm.json]: 👀 👍

**Pull request merged**